utils/dbt_format_utils.py

```diff
@@ -0,0 +1,329 @@
+            args.outputPath,
+        )
+
+    elif args.action == "restructAirbyteDbt":
+        # Check Args for Action
+        if not args.airbyteWorkspacePath:
+            raise Exception("Make sure you specify the --airbyteWorkspacePath argument")
+        if not args.dbtCtaPath:
+            raise Exception("Make sure you specify the --dbtCtaPath argument")
+        # Run selected action
+        restructure_airbyte_dbt(args.airbyteWorkspacePath, args.dbtCtaPath)
+    else:
+        raise Exception(
+            "Please enter a supported action -> renameBaseTables, createMatViews, getCacherScript, restructAirbyteDbt"
+        )
+
+
+# Function to grab Airbyte Generated dbt and move into CTA structured folder (✨Progify it✨)
+def restructure_airbyte_dbt(airbyte_workspace_path, dbt_cta_path):
+    # Mapping of Airbyte folders -> CTA folders
+    airbyte_cta_dir_mapping = {
+        "airbyte_ctes": "0_ctes",
+        "airbyte_tables": "1_cta_full_refresh",
+        "airbyte_incremental": "1_cta_incremental",
+    }
+
+    # Go through and move all files from Airbyte export into 'dbt-cta' directories
+    for airbyte_dir, cta_dir in airbyte_cta_dir_mapping.items():
+        # Get all sql files that should be moved for current mapping
+        files_to_move = glob(
+            f"{airbyte_workspace_path}/**/{airbyte_dir}/**/*.sql", recursive=True
+        )
+
+        if files_to_move:  # Only run if there were any files found
+            # Make sure target path is created
+            dbt_cta_path = (
+                dbt_cta_path if dbt_cta_path[-1] != "/" else dbt_cta_path[:-1]
+            )
+            target_path = f"{dbt_cta_path}/models/{cta_dir}"
+            os.makedirs(target_path, exist_ok=True)
+
+            # Move files to target path
+            for file_path in files_to_move:
+                file_name = file_path.split("/")[-1]
+                moved_file = move(src=file_path, dst=f"{target_path}/{file_name}")
+                print(f"Moved {file_path} -> {moved_file}")
+
+    # Now lets move over the sources.yml file
+    sources_file = glob(
+        f"{airbyte_workspace_path}/**/models/**/sources.yml", recursive=True
+    )
+    if sources_file:
+        move_source_file = move(
+            src=sources_file[0], dst=f"{dbt_cta_path}/models/sources.yml"
+        )
+        print(f"Moved {sources_file[0]} -> {dbt_cta_path}/models/sources.yml")
+    else:
+        print(f"Couldnt find the sources file in {airbyte_workspace_path}")
+
+    # Cool, files are moved so now lets modify the file content
+    modify_dbt_models(f"{dbt_cta_path}/models/0_ctes", "cte")
+    modify_full_refresh_models(f"{dbt_cta_path}/models/1_cta_full_refresh")
+    modify_dbt_models(f"{dbt_cta_path}/models/1_cta_incremental", "incremental")
+
+
+# Function to modify ctes and incremental models. For these its just gonna be
+# swapping out the config section, and changing source to cta.
+def modify_dbt_models(path_to_models, model_type):
```

Comment:
general request: Can we add docstrings to all the functions? I know I am a broken record here but chatgpt does this VERY fast, and it will make the code more readable/maintanable/etcetc

Reply:
yup can do 👍